security/src/main/java/io/micronaut/security/authentication/BlockingAuthenticationProvider.java

```diff
@@ -18,13 +18,14 @@
 import io.micronaut.core.annotation.Blocking;
 import io.micronaut.core.annotation.NonNull;
 import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.naming.Named;
 
 /**
  * Defines an authentication provider.
  * @since 4.5.0
  * @param <T> Request
  */
-public interface BlockingAuthenticationProvider<T> {
+public interface BlockingAuthenticationProvider<T> extends Named {
```

Comment:
I considered this, but my thought is what value does extending `Named` actually provide? It seems like an unnecessary extra burden on the user in this case. There is no constraint in Micronaut that forces them to use the implemented name in the `@Named` qualifier annotation, is there? Nor does it force them to add the qualifier?

Reply:
> It seems like an unnecessary extra burden on the user in this case. 

I think the burden is minimal

> There is no constraint in Micronaut that forces them to use the implemented name in the @Named qualifier annotation, is there? 
No

> Nor does it force them to add the qualifier?
No

However, together with the docs (where we show using the same name qualifier as in the `getName` method) is friendly reminder for you not to forget to add the name qualifier to the class.

Reply:
Ok, fair enough.